crates/uv-tool/src/lib.rs

```diff
@@ -56,6 +56,13 @@ pub enum Error {
     Serialization(#[from] toml_edit::ser::Error),
 }
 
+/// A Package identitifier
+#[derive(Debug, Clone)]
+pub struct ToolName {
+    pub name: PackageName,
+    pub suffix: Option<String>,
```

Comment:
This may be a problematic structure for `uv tool upgrade` etc., e.g., if the user specifies a tool name via the command line we won't know what's the package name and what's a suffix until sometime later?